lib/ui/painting/canvas.cc

```diff
@@ -432,12 +432,11 @@ void Canvas::drawShadow(const CanvasPath* path,
                                           elevation, transparentOccluder, dpr);
 }
 
-void Canvas::Clear() {
+void Canvas::Invalidate() {
+  if (dart_wrapper()) {
```

Comment:
Who else invalidates the canvas? The only owner would be a picture recorder which gets its own bespoke canvas. In case the caller tries to end recording multiple time, the isRecording check on the wrapper (PictureRecorder) would trip. Not sure when the Dart wrapper would ever be unset here. Seems like a redundant check.

Reply:
This is defensive coding intended to avoid the scenario described in https://github.com/flutter/flutter/issues/60319#issuecomment-650453028 where a Dart API invocation causes the `Canvas` to be GCed.

If the `PictureRecorder` Dart object is still holding its reference to the `Canvas` when the `endRecording` native method is called, then it would not be possible for the `Canvas` to be GCed inside the native method.  But I wanted to add the check so that `Invalidate` will still be safe if the order of operations in the Dart code changes.